### PR TITLE
[CVP-2252] Updates to tests for deprecated APIs check

### PR DESCRIPTION
**Pull request by dirgim**
Updates to unit tests for deprecated APIs check, allow running multiple CI unit tests with different data

* Updates to tests for deprecated APIs check
* Add default positive operator bundle unit test
* Add deprecated bundle CI unit tests
* Change CI unit tests to contain resource links

### CODE REVIEW

File: Dockerfiles/ci/run_tests.py [hunk 1]
```diff
@@ -7,33 +7,82 @@
...
+        operator_work_dir = "{}/example-bundle-default-positive".format(self.test_dir)
+        work_dir = "{}/example-workdir-v43-v44".format(self.test_dir)
+        exec_cmd = "ansible-playbook -vvv -i localhost, --connection local \
+                operator-test-playbooks/validate-operator-bundle.yml \
+                -e 'operator_work_dir={operator_work_dir}' \
+                -e 'work_dir={work_dir}'".format(operator_work_dir=operator_work_dir,
+                                                 work_dir=work_dir)
+        playbook_command = subprocess.run(exec_cmd, shell=True)
+        print(playbook_command)
+        self.assertTrue(playbook_command.returncode == 2)
```
Comment (lslebodn) [inline Dockerfiles/ci/run_tests.py:83]: Please add a check to expected error message in the output as addition to checking return code.

**Review by lslebodn — CHANGES_REQUESTED** [review]

File: Dockerfiles/ci/run_tests.py [hunk 2]
```diff
@@ -7,33 +7,82 @@
...
-        self.operator_dir = os.getenv('OPERATOR_DIR',
-                                      "/project/operator_dir/")
-        self.operator_work_dir = os.getenv('OPERATOR_WORK_DIR',
-                                           "/project/test_operator_work_dir/")
-        self.work_dir = os.getenv('WORK_DIR',
-                                  "/project/output/")
-        self.playbooks_dir = os.getenv('PLAYBOOKS_DIR',
-                                       "/project/operator-test-playbooks/")
-
-    def test_for_report_success_empty_alm_examples(self):
```
[the remark this answers is not in the document]
Comment (dirgim) [reply Dockerfiles/ci/run_tests.py:22]: This test case concerned the legacy tests and isn't relevant for our testing any more.

File: Dockerfiles/ci/run_tests.py (hunk 1 above)
Comment (dirgim) [reply Dockerfiles/ci/run_tests.py:95]: Ack, added the check for the error message.

File: Dockerfiles/ci/run_tests.py [hunk 3]
```diff
@@ -7,33 +7,82 @@
...
+        work_dir = "{}/example-workdir-v43-v44".format(self.test_dir)
+        exec_cmd = "ansible-playbook -vvv -i localhost, --connection local \
+                operator-test-playbooks/validate-operator-bundle.yml \
+                -e 'operator_work_dir={operator_work_dir}' \
+                -e 'work_dir={work_dir}'".format(operator_work_dir=operator_work_dir,
+                                                 work_dir=work_dir)
+        playbook_command = subprocess.run(exec_cmd, shell=True)
+        print(playbook_command)
+        self.assertTrue(playbook_command.returncode == 2)
+
```
[the remark this answers is not in the document]
Comment (dirgim) [reply Dockerfiles/ci/run_tests.py:97]: Ack, added the additional test case.

File: Dockerfiles/ci/run_tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lslebodn) [reply Dockerfiles/ci/run_tests.py:22]: Sure we do not run it for backported version of bundles but there are still some operator build directly in app_registry format in downstream. So test is still valid.

Or if it is not valid than not just test should be removed but also related(and unused) playbooks and roles

[the remark this answers is not in the document]
Comment (dirgim) [reply Dockerfiles/ci/run_tests.py:22]: Fine. I'll put it back.

File: Dockerfiles/ci/run_tests.py [hunk 4]
```diff
@@ -7,33 +7,108 @@
...
+        work_dir = "{}/example-workdir-invalid-ocp-version".format(self.test_dir)
+        exec_cmd = "ansible-playbook -vvv -i localhost, --connection local \
+                operator-test-playbooks/validate-operator-bundle.yml \
+                -e 'operator_work_dir={operator_work_dir}' \
+                -e 'work_dir={work_dir}'".format(operator_work_dir=operator_work_dir,
+                                                 work_dir=work_dir)
+        playbook_command = subprocess.run(exec_cmd, shell=True, capture_output=True)
+        print(playbook_command.returncode)
+        self.assertTrue(playbook_command.returncode == 2)
+        self.assertIn("Invalid semver in ocpfournine", playbook_command.stdout.decode("utf-8"))
```
[the remark this answers is not in the document]
Comment (dirgim) [reply Dockerfiles/ci/run_tests.py:133]: Ack, added the additional check

**Review by lslebodn — APPROVED** [review]
lgtm

**Review by samvarankashyap — APPROVED** [review]
lgtm

**Pull request merged**